delta.go

```diff
@@ -302,11 +302,16 @@ func (t *Table) LatestVersion() (int64, error) {
 	)
 	if errors.Is(err, storage.ErrObjectDoesNotExist) {
 		for {
-			o, err := t.Store.List(storage.NewPath("_delta_log"), objects)
-			if err != nil {
-				return -1, fmt.Errorf("list Delta log: %w", err)
+			for {
+				o, listErr := t.Store.List(storage.NewPath("_delta_log"), objects)
+				if listErr == nil {
+					objects = &o
+					break
+				} else if errors.Is(listErr, storage.ErrListObjects) {
+					continue
```

Comment:
Is this a potential infinite loop?

Reply:
Yeah that's a good call, I think it's a lot cleaner if retrying is deferred to `CommitLogStore()` instead of handling that here. I added some error checking to `tryCommitLogStore()` that results in returning out of `tryCommitLogStore()` if `LatestVersion()` returns a network request-related error.